### PR TITLE
8272520: Inline GenericTaskQueue::initialize() to the constructor

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple refactoring of GenericTaskQueue initialization API.

Test: tier1_gc

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8272520](https://bugs.openjdk.java.net/browse/JDK-8272520): Inline GenericTaskQueue::initialize() to the constructor

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Ivan Walulya](https://openjdk.java.net/census#iwalulya) (@walulyai - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5125/head:pull/5125` \
`$ git checkout pull/5125`

Update a local copy of the PR: \
`$ git checkout pull/5125` \
`$ git pull https://git.openjdk.java.net/jdk pull/5125/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5125`

View PR using the GUI difftool: \
`$ git pr show -t 5125`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5125.diff">https://git.openjdk.java.net/jdk/pull/5125.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/taskqueue.inline.hpp [hunk 1]
```diff
@@ -49,7 +49,8 @@ inline GenericTaskQueueSet<T, F>::~GenericTaskQueueSet() {
 }
 
 template<class E, MEMFLAGS F, unsigned int N>
-inline void GenericTaskQueue<E, F, N>::initialize() {
+inline GenericTaskQueue<E, F, N>::GenericTaskQueue() : _last_stolen_queue_id(InvalidQueueId), _seed(17 /* random number */) {
+  assert(sizeof(Age) == sizeof(size_t), "Depends on this.");
   _elems = ArrayAllocator<E>::allocate(N, F);
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/taskqueue.inline.hpp:54]: `_elems` could be initialized in the mem-initializer-list.

File: src/hotspot/share/gc/shared/taskqueue.inline.hpp [hunk 2]
```diff
@@ -49,7 +49,8 @@ inline GenericTaskQueueSet<T, F>::~GenericTaskQueueSet() {
 }
 
 template<class E, MEMFLAGS F, unsigned int N>
-inline void GenericTaskQueue<E, F, N>::initialize() {
+inline GenericTaskQueue<E, F, N>::GenericTaskQueue() : _last_stolen_queue_id(InvalidQueueId), _seed(17 /* random number */) {
+  assert(sizeof(Age) == sizeof(size_t), "Depends on this.");
```
Comment (kimbarrett) [inline src/hotspot/share/gc/shared/taskqueue.inline.hpp:53]: [pre-existing] This assert is over-constraining.  The real requirement is static-asserted in the Age class.

**Review by kimbarrett — APPROVED** [review]
Looks good.  Just a couple minor nits.

File: src/hotspot/share/gc/shared/taskqueue.inline.hpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/taskqueue.inline.hpp]: Fixed.

File: src/hotspot/share/gc/shared/taskqueue.inline.hpp (hunk 2 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/shared/taskqueue.inline.hpp]: Removed.

**Review by kimbarrett — APPROVED** [review]
Looks even better.

**Review by walulyai — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**